src/modules/menu/components/share/MenuShareSection.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+    <MenuSection
+        :title="$t('share')"
+        data-cy="menu-share-section"
+        secondary
+        @click.once="generateShortLinkIfMNeeded"
```

Comment:
That will probably lead to a ticket down the road. I'm not really familiar with the Vue router. Could we save a kind of snapshot of the routing state and then use that to check if we need to regenerate the short-link?